client-mr/src/main/java/org/apache/hadoop/mapreduce/v2/app/RssMRAppMaster.java

```diff
@@ -366,7 +366,10 @@ static void writeExtraConf(JobConf conf, JobConf extraConf) {
       long size = status.getLen();
       String sizes = conf.get(MRJobConfig.CACHE_FILES_SIZES);
       conf.set(MRJobConfig.CACHE_FILES_SIZES, sizes == null ? String.valueOf(size) : size + "," + sizes);
-    } catch (Exception e) {
+    } catch (InterruptedException e) {
+      LOG.error("Interrupted while trying to upload extra conf ", e);
+      Thread.currentThread().interrupt();
```

Comment:
https://stackoverflow.com/questions/3976344/handling-interruptedexception-in-java

Reply:
> By now it should be clear that just doing throw new RuntimeException(e) is a bad idea. It isn't very polite to the caller. You could invent a new runtime exception but the root cause (someone wants the thread to stop execution) might get lost.

Quoted from the link above, it's a bad idea to throw new RuntimeException(e).
And we cannot propagate InterruptedException here because `RssMRAppMaster` extends `org.apache.hadoop.mapreduce.v2.app.MRAppMaster`.

Reply:
We could check `Thread.currentThread().isInterrupted()` in `main()` or just call `System.exit()` here.